test/models/thesis_test.rb

```diff
@@ -1245,4 +1245,100 @@ def attach_file_with_purpose_to(thesis, purpose = 'thesis_pdf')
+    proquest_consent_count = Thesis.consented_to_proquest.count
+    assert proquest_consent_count < Thesis.count
+
+    # single-author thesis that has opted in is included
+    thesis = theses(:one)
+    assert_includes Thesis.consented_to_proquest, thesis
+
+    # single-author thesis that has opted out is excluded
+    thesis = theses(:with_note)
+    assert_not_includes Thesis.consented_to_proquest, thesis
+
+    # single-author thesis with no opt-in status is excluded
+    thesis = theses(:coauthor)
+    assert_not_includes Thesis.consented_to_proquest, thesis
+
+    # multi-author thesis that has opted in is included
+    thesis = theses(:two)
+    assert_includes Thesis.consented_to_proquest, thesis
+
+    # multi-author thesis that has opted out is excluded
+    thesis = theses(:with_hold)
+    assert_not_includes Thesis.consented_to_proquest, thesis
+
+    # multi-author thesis with conflicting opt-in statuses is excluded
+    thesis = theses(:doctor)
+    assert_not_includes Thesis.consented_to_proquest, thesis
+  end
+
+  test 'invalid without proquest_exported' do
+    thesis = theses(:one)
+    assert thesis.proquest_exported = 'Not exported'
+    assert thesis.valid?
+
+    thesis.proquest_exported = 'Full harvest'
+    thesis.save
+    assert thesis.valid?
+
+    thesis.proquest_exported = 'Partial harvest'
+    thesis.save
+    assert thesis.valid?
+  end
+
+  test 'not_consented_to_proquest returns only theses with authors that have not opted in to ProQuest' do
+    # single-author thesis that has opted out is included
+    thesis = theses(:with_note)
+    assert_includes Thesis.not_consented_to_proquest, thesis
+
+    # single-author thesis with no opt-in status is included
+    thesis = theses(:coauthor)
+    assert_includes Thesis.not_consented_to_proquest, thesis
+
+    # single-author thesis that has opted in is excluded
+    thesis = theses(:one)
+    assert_not_includes Thesis.not_consented_to_proquest, thesis
+
+    # multi-author thesis with all opt-outs is included
+    thesis = theses(:with_hold)
+    assert_includes Thesis.not_consented_to_proquest, thesis
+
+    # multi-author thesis with one opt-in and one opt-out is excluded
+    thesis = theses(:doctor)
+    assert_not_includes Thesis.not_consented_to_proquest, thesis
+
+    # multi-author thesis that has opted in is excluded
+    thesis = theses(:two)
+    assert_not_includes Thesis.not_consented_to_proquest, thesis
+  end
+
+  test 'exported_to_proquest includes partially harvested theses' do
+    partially_harvested_thesis = theses(:doctor)
+    assert_includes Thesis.exported_to_proquest, partially_harvested_thesis
+  end
+
+  test 'exported_to_proquest includes fully harvested theses' do
+    fully_harvested_thesis = theses(:engineer)
+    assert_includes Thesis.exported_to_proquest, fully_harvested_thesis
+  end
+
+  test 'exported_to_proquest excludes theses that have not been harvested' do
```

Comment:
The wording on this (and possibly the two previous tests but I haven't decided) as well as the next few may be awkward. We are only in control of when we export theses to Proquest not when they actually get around to Harvesting them. The enum values themselves are fine as we export them as "Full harvest"able.

My concern is that in the future these tests names may imply that we somehow are updating/knowing when Proquest comes back and harvests theses which is not the case.

Rewording to something like `exported_to_proquest excludes theses that are flagged not harvestable` (I'm not sure if that is entirely correct wording but I hope it provides a useful example of what I'm trying to convey)

Reply:
Agreed. I'll clarify the language in these.